common-util/src/main/kotlin/com/google/devtools/ksp/processing/impl/CodeGeneratorImpl.kt

```diff
@@ -154,18 +155,25 @@ class CodeGeneratorImpl(
         associate(sources, file)
     }
 
+    private val File.relativeFile: File
+        get() =
+            if (this.startsWith(buildDir))
```

Comment:
I found the reason this fails on Windows.

Windows uses counter slash for directory separator, but in Java API it seems to be intepreting the path as URI, which uses normal slash as directory separator, and this causes lots of issues that will break simple string check.

One workaround is to manually replace counter slash with normal slash when doing check, but counter slash is a valid file name character on unix, uncommon but valid, which messes things up again... 

Maybe also put an OS check before checking the strings, but that looks just too bad as a code to me.